### PR TITLE
Remove "default" behaviour from spec, and clarify use of auto

### PR DESCRIPTION
**Pull request by simonrankine**
Resolves issue #38 by removing the "default" values from the spec, as they never actually had a meaningful use anyway. Also clarified the use of auto, as per #38.

### CODE REVIEW

File: APIs/ConnectionAPI.raml [hunk 1]
```diff
@@ -254,13 +254,16 @@ documentation:
...
-               into the actual values that will be used by the sender. For example if a sender has the `destination_ip`
+               Get active sender parameters. On activation all instances of "auto" must be resolved
+               into the actual values that will be used by the sender, unless there is an error condition.
+               For example if a sender has the `destination_ip`
                parameter set to `auto` in /staged then `destination_ip` in /active should show the multicast IP
                chosen by the receiver. The parameter will remain set to `auto` in staged, and may be resolved to a different
                value on each subsequent activation. In cases, such as `destination_ip`, where the parameter changing unexpectedly
                at activation through re-resolution may be undesirable, care should be taken in implementation
                to manage this.
+               If there is an error condition that means `auto` cannot be resolved the active transport paraemters
```
Comment (andrewbonney) [inline APIs/ConnectionAPI.raml]: Minor typo (parameters) and possible missing comma before 'the active'

File: APIs/ConnectionAPI.raml [hunk 2]
```diff
@@ -442,13 +445,16 @@ documentation:
...
-             into the actual values that will be used by the receiver. For example if a receiver has the `interface_ip`
+             Get active receiver parameters. On activation all instances of "auto" must be resolved
+             into the actual values that will be used by the receiver, unless there is an error condition.
+             For example if a receiver has the `interface_ip`
              parameter set to `auto` in /staged then `interface_ip` in /active should show the IP address of the interface
              chosen by the receiver. The parameter will remain set to `auto` in staged, and may be resolved to a different
              value on each subsequent activation. In cases, such as `interface_ip`, where the parameter changing unexpectedly
              at activation through re-resolution may be undesirable, care should be taken in implementation
              to manage this.
+             If there is an error condition that means `auto` cannot be resolved the active transport paraemters
```
Comment (andrewbonney) [inline APIs/ConnectionAPI.raml]: Same issue as above

File: APIs/schemas/v1.0_sender_transport_params_rtp.json [hunk 3]
```diff
@@ -71,44 +67,36 @@
           },
           {
             "format": "ipv6"
-          },
-          {
-            "pattern": "^auto$"
```
Comment (andrewbonney) [inline APIs/schemas/v1.0_sender_transport_params_rtp.json]: Was removing this option intentional?

**Review by andrewbonney — COMMENTED** [review]
Looks fine to me. Just a couple of minor typos and a small query.

**Review by peterbrightwell — APPROVED** [review]
Following discussion with Simon, I'm happy with this.

**Pull request merged**